services/blockchain-indexer/shared/indexer/blockchainIndex.js

```diff
@@ -512,38 +549,11 @@ const deleteIndexedBlocks = async job => {
 	}
 };
 
-let isIndexingRunning = false;
-const BLOCKCHAIN_INDEX_RESCHEDULE_DELAY = 1000;
-
-const indexBlockAtomicWrapper = async job => {
-	if (isIndexingRunning) {
-		logger.trace('Already indexing another block!');
-		await delay(BLOCKCHAIN_INDEX_RESCHEDULE_DELAY);
-		// eslint-disable-next-line no-use-before-define
-		await addHeightToIndexBlocksQueue(job.data.height);
-		return;
-	}
-
-	isIndexingRunning = true;
-
-	try {
-		await indexBlock(job);
-	} catch (err) {
-		isIndexingRunning = false;
-		logger.error(
-			`Error occurred during indexing block.\nError: ${err.message}\nStack:${err.stack}`,
-		);
-		throw new Error(err);
-	}
-
-	isIndexingRunning = false;
-};
-
```

Comment:
since we are using `upsert` operations, we could end up having mismatched data without the serialized wrapper. consider this scenario-

last height indexed is 100. now the next job will be indexing block with height 101. Both of the indexing job are executed simultaneously by the queue and we found out that the last indexed height is 100. so both will try to commit 101th block and both will succeed as this are `upsert` operation.